### PR TITLE
Documentation regarding tap gesture callbacks

### PR DESCRIPTION
**Pull request by Hixie**
cc @gspencergoog

### CODE REVIEW

File: packages/flutter/lib/src/gestures/tap.dart [hunk 1]
```diff
@@ -61,26 +91,73 @@ typedef void GestureTapCancelCallback();
 /// pointer interactions during a tap sequence are not recognized as additional
 /// taps. For example, down-1, down-2, up-1, up-2 produces only one tap on up-1.
 ///
+/// The lifecycle of events for a tap gesture are as follows:
```
Comment (gspencergoog) [inline packages/flutter/lib/src/gestures/tap.dart]: are --> is

File: packages/flutter/lib/src/gestures/tap.dart [hunk 2]
```diff
@@ -61,26 +91,73 @@ typedef void GestureTapCancelCallback();
...
+///
+/// * [onTapDown], which triggers after a short timeout ([deadline]) even if the
+///   gesture has not won its arena yet.
```
Comment (gspencergoog) [inline packages/flutter/lib/src/gestures/tap.dart]: You might explain what an arena is, or link to an appropriate class. It's not that clear what is meant by arena if this is the first gesture class you encounter.

**Review by gspencergoog — APPROVED** [review]

File: packages/flutter/lib/src/gestures/tap.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/gestures/tap.dart]: fixed

File: packages/flutter/lib/src/gestures/tap.dart (hunk 2 above)
Comment (Hixie) [reply packages/flutter/lib/src/gestures/tap.dart]: done

**Comment by Hixie** [comment]
will land on green

**Comment by Hixie** [comment]
@fkorotkov Hard to say if it's our problem or yours, but the tests-windows shard here spun around doing nothing for like an hour. Then it did some more stuff and then eventually timed out. At a guess, there's some sort of resource starvation going on in the Windows cluster.

**Comment by fkorotkov** [comment]
@Hixie me and @gspencergoog saw that but there were issues with reporting logs for such timed out tasks. I fixed it yesterday and at least now there are logs presented every time (they were missing sometimes for timed out tasks).

It seems it was hanging on running `slivers_evil_test.dart`. 

If you are interested I can pretty easy add a way to run windows containers on GCP instead of Azure. The only problem will be that Windows Contianers on GCP will start in 4-5 minutes instead of 1:30-2 on Azure. But we can check if it will be more stable.

**Comment by Hixie** [comment]
@fkorotkov I ran it again and this time it was 60 minutes on a different test (`C:\Windows\Temp\flutter sdk\packages\flutter_localizations\test\date_picker_test.dart: DayPicker shows dates for ar_AR`). I'll rerun it again.

Log of that failing run: [5694495647596544-test_all.log](https://github.com/flutter/flutter/files/2295837/5694495647596544-test_all.log)

**Pull request merged**